lib/psych/scalar_scanner.rb

```diff
@@ -4,6 +4,8 @@ module Psych
   ###
   # Scan scalars for built in types
   class ScalarScanner
+    autoload :Date, "date"
```

Comment:
I suppose this usage of autoload is debatable, I've seen it elsewhere, but I can change it for a `require unless defined?(Date)` if you prefer.